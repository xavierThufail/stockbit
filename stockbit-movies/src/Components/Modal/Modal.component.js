import React from 'react';
import { Modal as AntDesignModal, Col, Row } from 'antd';
import { CloseOutlined } from '@ant-design/icons';

import './Modal.component.css';
import { TouchableOpacity } from '../index';
import config from './Modal.config';
import styles from './Modal.style';

const renderCloseButton = ({ handleClose }) => (
  <TouchableOpacity onClick={handleClose} style={styles.btnClose}>
    <CloseOutlined />
  </TouchableOpacity>
);

const renderTitleModal = ({ title, style }) => (
  <div style={styles.titleModal}>
    {title}
  </div>
);

const renderContent = ({ children, handleClose, titleStyle, title }) => (
  <Row>
    <Col style={styles.col}>
      {renderCloseButton({ handleClose })}
      {renderTitleModal({ title, style: titleStyle })}
      {children}
    </Col>
  </Row>
);

const Modal = ({ children, handleClose, visible, titleStyle, title, destroyOnClose, style, url }) => (
  <AntDesignModal
    title={false}
    visible={visible}
    closable={false}
    footer={null}
    bodyStyle={styles.content(url)}
    width={300}
    centered
    destroyOnClose={destroyOnClose}
    style={style}
  >
    {renderContent({ children, handleClose, titleStyle, title })}
  </AntDesignModal>
);

Modal.propTypes = config.propTypes;
Modal.defaultProps = config.defaultProps;
Modal.displayName = config.displayName;

export default Modal;
