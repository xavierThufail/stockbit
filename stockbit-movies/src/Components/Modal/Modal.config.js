import PropTypes from 'prop-types';

const propTypes = {
  children: PropTypes.node,
  handleClose: PropTypes.func.isRequired,
  visible: PropTypes.bool.isRequired,
  titleStyle: PropTypes.object,
  title: PropTypes.string.isRequired,
  destroyOnClose: PropTypes.bool
};

const defaultProps = {
  titleStyle: {},
  destroyOnClose: false
};

const displayName = 'Modal';

export default {
  propTypes,
  defaultProps,
  displayName
};
