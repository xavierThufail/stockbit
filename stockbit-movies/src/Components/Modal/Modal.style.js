const styles = {
  btnClose: {
    position: 'absolute',
    right: 0,
    fontSize: '20px',
    lineHeight: 1,
    zIndex: 1000,
    borderRadius: '50px',
    backgroundColor: '#191a1f',
    color: '#c33e32',
    fontWeight: 'bold',
    padding: '5px'
  },
  content: (url) => ({
    height: '450px',
    backgroundImage: `url(${url})`,
    backgroundPosition: 'center',
    backgroundRepeat: 'no-repeat',
    borderRadius: '10px',
  }),
  col: {
    width: '100%',
    position: 'relative'
  },
  titleModal: {
    textAlign: 'center',
    fontSize: '24px',
    marginBottom: '40px',
    fontWeight: 'bold',
    color: '#00a4de',
    lineHeight: 1
  }
};

export default styles;
