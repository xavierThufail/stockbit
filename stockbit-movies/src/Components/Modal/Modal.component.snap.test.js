import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { renderHook } from '@testing-library/react-hooks';

import Modal from './Modal.component';

describe('Modal component snap test', () => {
  it('Render Modal, should match snapshot', () => {
    const mockedProps = {
      children: (<div>Test</div>),
      handleClose: jest.fn(),
      visible: true,
      title: 'Title'
    };

    const { result } = renderHook(() => new Modal(mockedProps));

    expect(result.current).toMatchSnapshot();
  });
});
