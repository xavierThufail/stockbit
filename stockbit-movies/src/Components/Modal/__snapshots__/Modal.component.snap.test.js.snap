// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component snap test Render Modal, should match snapshot 1`] = `
<Modal
  bodyStyle={
    Object {
      "padding": 30,
    }
  }
  centered={true}
  closable={false}
  confirmLoading={false}
  footer={null}
  maskTransitionName="fade"
  okType="primary"
  title={false}
  transitionName="zoom"
  visible={true}
  width={450}
>
  <Row>
    <Col
      style={
        Object {
          "position": "relative",
          "width": "100%",
        }
      }
    >
      <TouchableOpacity
        onClick={[MockFunction]}
        style={
          Object {
            "fontSize": "30px",
            "lineHeight": 1,
            "position": "absolute",
            "right": 0,
            "zIndex": 1000,
          }
        }
      >
        <ForwardRef(CloseOutlined) />
      </TouchableOpacity>
      <div
        style={
          Object {
            "color": "#00a4de",
            "fontSize": "24px",
            "fontWeight": "bold",
            "lineHeight": 1,
            "marginBottom": "40px",
            "textAlign": "center",
          }
        }
      >
        Title
      </div>
      <div>
        Test
      </div>
    </Col>
  </Row>
</Modal>
`;
